diff --git a/Makefile.am b/Makefile.am
index 09ff61e..eebb8ce 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,7 @@ ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 SUBDIRS = help po
 
 DISTCHECK_CONFIGURE_FLAGS = \
-	--enable-appstream-util \
+	--enable-appdata-validate \
 	--enable-compile-warnings=maximum \
 	--enable-rdp
 
@@ -256,9 +256,9 @@ nodist_desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 mimedir = $(datadir)/mime/packages
 nodist_mime_DATA = data/vinagre-mime.xml
 
-@APPSTREAM_XML_RULES@
-appstream_in_files = data/vinagre.appdata.xml.in
-appstream_XML = $(appstream_in_files:.appdata.xml.in=.appdata.xml)
+@APPDATA_XML_RULES@
+appdata_in_files = data/vinagre.appdata.xml.in
+appdata_XML = $(appdata_in_files:.appdata.xml.in=.appdata.xml)
 
 gsettings_SCHEMAS = data/org.gnome.Vinagre.gschema.xml
 
@@ -296,7 +296,7 @@ dist-changelog:
 	else rm -f .ChangeLog.tmp; exit 1; fi
 
 dist_noinst_DATA = \
-	$(appstream_in_files) \
+	$(appdata_in_files) \
 	$(noinst_vinagreh_headers) \
 	$(vinagre_vala_sources) \
 	vinagre/vinagre-vala.h \
@@ -314,7 +314,7 @@ dist_noinst_DATA = \
 	vinagre.doap
 
 CLEANFILES = \
-	$(appstream_XML) \
+	$(appdata_XML) \
 	$(nodist_desktop_DATA) \
 	$(nodist_mime_DATA) \
 	$(nodist_pkgconfig_DATA) \
diff --git a/configure.ac b/configure.ac
index 0d3344d..2a351c1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -214,7 +214,7 @@ YELP_HELP_INIT
 
 GLIB_GSETTINGS
 
-APPSTREAM_XML
+APPDATA_XML
 
 AC_CONFIG_FILES([
 Makefile
