--- linux_lib/libcrystalhd/Makefile.orig	2013-10-23 12:34:15.156574046 +0100
+++ linux_lib/libcrystalhd/Makefile	2013-10-23 12:34:24.639906696 +0100
@@ -24,8 +24,8 @@
 # -D_USE_SHMEM_
 
 CPPFLAGS += ${INCLUDES}
-CPPFLAGS += -O2 -Wall -fPIC -shared -fstrict-aliasing -msse2
-LDFLAGS = -Wl,-soname,${BCLIB_SL} -pthread
+CXXFLAGS += -O2 -Wall -fPIC -shared -fstrict-aliasing -msse2
+LDFLAGS += -Wl,-soname,${BCLIB_SL} -pthread
 
 SRCFILES = 	libcrystalhd_if.cpp \
 		libcrystalhd_int_if.cpp \
@@ -38,7 +38,7 @@
 OBJFILES = ${SRCFILES:.cpp=.o}
 
 all:help $(OBJFILES)
-	$(BCGCC) $(CPPFLAGS) $(LDFLAGS) -o $(BCLIB) ${OBJFILES}
+	$(BCGCC) $(CPPFLAGS) $(CXXFLAGS) $(LDFLAGS) -o $(BCLIB) ${OBJFILES}
 	ln -sf $(BCLIB) $(BCLIB_NAME)
 	ln -sf $(BCLIB) $(BCLIB_SL)
 
