--- lib/upnp.c.orig	2016-02-15 18:15:17.533307109 +0000
+++ lib/upnp.c	2016-02-15 18:13:29.600628701 +0000
@@ -155,6 +155,9 @@
                            NULL,
                            0
 #if HAVE_LIBMINIUPNP17
+#if HAVE_LIBMINIUPNP110
+                          ,0
+#endif/* HAVE_LIBMINIUPNP17 */
                           ,0,
                           &rc
 #endif /* HAVE_LIBMINIUPNP17 */
--- miniupnpc.m4.orig	2016-02-15 18:15:31.466986223 +0000
+++ miniupnpc.m4	2016-02-15 18:05:25.381947682 +0000
@@ -37,6 +37,7 @@
          [AC_MSG_RESULT(yes)
           AC_DEFINE(HAVE_LIBMINIUPNP, 1, [UPNP support library])
           AC_DEFINE(HAVE_LIBMINIUPNP17, 1, [UPNP support library 1.7])
+          AC_DEFINE(HAVE_LIBMINIUPNP110, 1, [UPNP support library 1.10])
           unset no_upnp
 	  SOCKDDEPS="${SOCKDDEPS}${SOCKDDEPS:+ }$UPNPLIB"
 	  DLIBDEPS="${DLIBDEPS}${DLIBDEPS:+ }$UPNPLIB"],
