--- configure.ac~	2014-08-31 17:27:27.000000000 +0100
+++ configure.ac	2014-08-31 17:50:26.013322550 +0100
@@ -83,22 +83,22 @@
    AS_HELP_STRING([--with-libsoup-system], [Use libsoup system library to build(default=no)]),
    [with_libsoup_system=$withval], [with_libsoup_system=no])
 if test "$with_libsoup_system" = "no"; then
-    GLIB_REQUIRED=2.38.0
-    AM_PATH_GLIB_2_0($GLIB_REQUIRED,,,gobject gio)
-    if test "$GLIB_LIBS" = ""; then
-       AC_MSG_ERROR(GLIB $GLIB_REQUIRED or later is required to build libsoup)
-    fi
-    GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_38"
-
-    GLIB_MAKEFILE='$(top_srcdir)/Makefile.glib'
-    AC_SUBST(GLIB_MAKEFILE)
-
     PKG_CHECK_MODULES(XML, libxml-2.0)
     AC_SUBST(XML_CFLAGS)
     AC_SUBST(XML_LIBS)
 fi
 AM_CONDITIONAL(BUILD_LIBSOUP, test $with_libsoup_system = no)
 
+GLIB_REQUIRED=2.38.0
+AM_PATH_GLIB_2_0($GLIB_REQUIRED,,,gobject gio)
+if test "$GLIB_LIBS" = ""; then
+   AC_MSG_ERROR(GLIB $GLIB_REQUIRED or later is required to build libsoup)
+fi
+GLIB_CFLAGS="$GLIB_CFLAGS -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_38"
+
+GLIB_MAKEFILE='$(top_srcdir)/Makefile.glib'
+AC_SUBST(GLIB_MAKEFILE)
+
 GLIB_TESTS
 
 PKG_CHECK_MODULES(SQLITE, sqlite3)
