--- configure.ac~	2016-01-19 22:13:59.000000000 +0000
+++ configure.ac	2016-01-19 22:33:01.590486365 +0000
@@ -525,7 +525,7 @@
   AS_IF([test "x$ARG_ENABLE_FULL_EXPORT" = "xno"], [
      CXXFLAGS="$CXXFLAGS -fvisibility=hidden -fvisibility-inlines-hidden"
      CFLAGS="$CFLAGS -fvisibility=hidden"
-     LDFLAGS="$LDFLAGS -fvisibility=hidden -fvisibility-inlines-hidden"
+     LDFLAGS="$LDFLAGS -fvisibility=hidden"
   ])
 ])
 
