--- configure.ac.orig	2014-09-17 00:39:06.649266820 +0100
+++ configure.ac	2014-09-17 00:46:49.631031011 +0100
@@ -288,6 +288,15 @@
 )
 
 AC_ARG_ENABLE(
+  [i2p],
+  [AS_HELP_STRING(
+    [--enable-i2p],
+    [enable I2P support [default=no]])],
+  [[ARG_ENABLE_I2P=$enableval]],
+  [[ARG_ENABLE_I2P=no]]
+)
+
+AC_ARG_ENABLE(
   [examples],
   [AS_HELP_STRING(
     [--enable-examples],
@@ -514,6 +523,20 @@
    AC_MSG_ERROR([Unknown option "$ARG_ENABLE_POOL_ALLOC". Use either "yes" or "no".])]
 )
 
+AC_MSG_CHECKING([whether I2P support should be enabled])
+AS_CASE(["$ARG_ENABLE_I2P"],
+  ["yes"], [
+    AC_MSG_RESULT([yes])
+    AC_DEFINE([TORRENT_USE_I2P],[1],[Define to enable the I2P support.])
+    COMPILETIME_OPTIONS="$COMPILETIME_OPTIONS -DTORRENT_USE_I2P=1 "
+   ],
+  ["no"], [
+    AC_MSG_RESULT([no])
+   ],
+  [AC_MSG_RESULT([$ARG_ENABLE_I2P])
+   AC_MSG_ERROR([Unknown option "$ARG_ENABLE_I2P". Use either "yes" or "no".])]
+)
+
 AS_ECHO
 AS_ECHO "Checking for extra build files:"
 
@@ -739,6 +762,7 @@
   geoip support:        ${ARG_ENABLE_GEOIP:-yes}
   dht support:          ${ARG_ENABLE_DHT:-yes}
   pool allocators:      ${ARG_ENABLE_POOL_ALLOC:-yes}
+  i2p support:          ${ARG_ENABLE_I2P:-no}
 
 Extra builds:
   examples:             ${ARG_ENABLE_EXAMPLES:-no}
