--- src/http_tracker_connection.cpp~	2014-09-18 07:51:20.000000000 +0100
+++ src/http_tracker_connection.cpp	2014-09-18 08:21:05.520253778 +0100
@@ -494,16 +494,6 @@
 		lazy_entry const* ipv6_peers = 0;
 #endif
 
-		// if we didn't receive any peers. We don't care if we're stopping anyway
-		if (peers_ent == 0 && ipv6_peers == 0
-			&& tracker_req().event != tracker_request::stopped)
-		{
-			fail(error_code(errors::invalid_peers_entry), -1, ""
-				, interval, min_interval);
-			return;
-		}
-
-
 		// look for optional scrape info
 		address external_ip;
 
