--- ./Source/Core/InputCommon/ControllerInterface/ControllerInterface.h~	2017-02-16 21:54:55.000000000 +0000
+++ ./Source/Core/InputCommon/ControllerInterface/ControllerInterface.h	2017-02-16 22:01:34.245298934 +0000
@@ -9,6 +9,7 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <functional>
 
 #include "Common/CommonTypes.h"
 #include "InputCommon/ControllerInterface/Device.h"
