--- Source/Core/Core/NetPlayServer.cpp.orig	2016-07-24 15:45:22.576344052 +0100
+++ Source/Core/Core/NetPlayServer.cpp	2016-07-24 15:45:50.520116495 +0100
@@ -1022,7 +1022,7 @@
   {
     std::unique_ptr<char, decltype(&std::free)> descXML(nullptr, std::free);
     int statusCode = 200;
-#if MINIUPNPC_API_VERSION >= 16
+#if MINIUPNPC_API_VERSION >= 14
     descXML.reset(static_cast<char*>(
         miniwget_getaddr(dev->descURL, &descXMLsize, cIP, sizeof(cIP), 0, &statusCode)));
 #else
