--- ./libraries/source/spidermonkey/build.sh~	2015-11-27 12:37:03.000000000 +0000
+++ ./libraries/source/spidermonkey/build.sh	2015-11-27 17:53:08.984088121 +0000
@@ -97,7 +97,7 @@
 perl -i.bak -pe 's/(LIBRARY_NAME\s+=).*/$1 '\''mozjs31-ps-release'\''/' moz.build
 mkdir -p build-release
 cd build-release
-CXXFLAGS="${TLCXXFLAGS}" ../configure ${CONF_OPTS} --with-nspr-libs="$NSPR_LIBS" --with-nspr-cflags="$NSPR_INCLUDES" --enable-optimize  # --enable-gczeal --enable-debug-symbols
+CXXFLAGS="${TLCXXFLAGS} ${CXXFLAGS}" ../configure ${CONF_OPTS} --with-nspr-libs="$NSPR_LIBS" --with-nspr-cflags="$NSPR_INCLUDES" --enable-optimize  # --enable-gczeal --enable-debug-symbols
 ${MAKE} ${MAKE_OPTS}
 cd ..
 
