--- Makefile	2014-12-09 15:40:11.000000000 +0000
+++ /tmp/Makefile	2015-11-15 19:30:20.581621768 +0000
@@ -13,17 +13,19 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # General Public License for more details.
 
+CC := gcc
+
 ccflags-y = -Wall
 
 all: i8kctl probe_i8k_calls_time
 
 i8kctl: i8kctl.c i8kctl.o
-	gcc -Wall i8kctl.c -o i8kctl
+	${CC} ${CFLAGS} i8kctl.c -o i8kctl
 
-probe_i8k_calls_time: probe_i8k_calls_time.c
-	gcc -Wall -c -g -DLIB i8kctl.c
-	gcc -Wall -c -g -DLIB probe_i8k_calls_time.c
-	gcc -o probe_i8k_calls_time i8kctl.o probe_i8k_calls_time.o
+probe_i8k_calls_time: probe_i8k_calls_time.c i8kctl.c
+	${CC} -c -Wall -fgnu89-inline $(CFLAGS) -DLIB probe_i8k_calls_time.c
+	${CC} -c -Wall $(CFLAGS) -DLIB i8kctl.c
+	${CC} ${LDFLAGS} i8kctl.o probe_i8k_calls_time.o -o probe_i8k_calls_time
 
 i8k:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
