--- ./src/lib/emile/emile_cipher_gnutls.c.orig	2016-09-09 08:35:58.709906791 +0000
+++ ./src/lib/emile/emile_cipher_gnutls.c	2016-09-09 08:39:55.636904136 +0000
@@ -168,8 +168,8 @@
 EAPI Eina_Bool
 emile_binbuf_sha1(const Eina_Binbuf * data, unsigned char digest[20])
 {
-   Eina_Slice slice = eina_binbuf_slice_get(data);
-   return emile_sha1(data.mem, data.len, digest);
+//   Eina_Slice slice = eina_binbuf_slice_get(data);
+   return emile_sha1(eina_binbuf_string_get(data), eina_binbuf_length_get(data), digest);
 }
 
 EAPI Eina_Binbuf *
