From a44a334feb29c8326478032eb3b6e1fc0ee96cbc Mon Sep 17 00:00:00 2001
From: Bertrand Jacquin <bertrand@jacquin.bzh>
Date: Fri, 10 Jan 2014 21:31:36 +0100
Subject: [PATCH 1/6] MEDIUM/BUILD: Use CC instead of LD to link library

---
 Makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index e0b69dc..a992458 100644
--- a/Makefile
+++ b/Makefile
@@ -27,7 +27,6 @@ SPEC_LDFLAGS	= \
 	-lcrypto
 
 CC	= cc
-LD	= $(CC)
 
 STRIP	= strip
 STRIP_FLAGS	= --strip-unneeded -R .comment -R .GCC.command.line -R .note.gnu.build-id
@@ -54,8 +53,8 @@ dist:	$(NAME)-$(VERSION).txz
 	$(Q)$(CC) -c $(SPEC_CFLAGS) $(CFLAGS) -o $@ $<
 
 $(NAME).so:	$(SRC:c=o)
-	$(call qcmd,LD,$@)
-	$(Q)$(LD) -shared -o $@ $^ $(SPEC_LDFLAGS) $(LDFLAGS)
+	$(call qcmd,CC,$@)
+	$(Q)$(CC) -shared -o $@ $^ $(SPEC_LDFLAGS) $(LDFLAGS)
 
 $(NAME).so.strip:       $(NAME).so
 	$(call qcmd,STRIP,$@)
-- 
1.8.3.2

