From 822eee883be620fd92e7e4431abbc68e81a88699 Mon Sep 17 00:00:00 2001
From: Bertrand Jacquin <bertrand@jacquin.bzh>
Date: Fri, 10 Jan 2014 21:40:46 +0100
Subject: [PATCH 4/6] MINOR/BUILD: Rename EXTRA_FILES to DOC_FILES

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 47853b4..c24c858 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@ SRC	= \
 	ciphers.c \
 	digests.c
 
-EXTRA_FILES	= \
+DOC_FILES	= \
 	README
 
 # User defined CFLAGS
@@ -65,7 +65,7 @@ $(NAME).so.strip:       $(NAME).so
 	$(call qcmd,STRIP,$@)
 	$(Q)$(STRIP) $(STRIP_FLAGS) -o $@ $<
 
-$(NAME)-$(VERSION).tar: $(SRC) $(EXTRA_FILES) Makefile
+$(NAME)-$(VERSION).tar: $(SRC) $(DOC_FILES) Makefile
 	$(call qcmd,TAR,$@)
 	$(Q)$(TAR) cf $@ \
 		--transform "s,^,$(NAME)-$(VERSION)/," \
-- 
1.8.3.2

