--- ftjam-2.5.3rc2-orig/jambase.c
+++ ftjam-2.5.3rc2/jambase.c
@@ -650,7 +650,7 @@
 "DC ?= dmd ;\n",
 "\n",
 "\n",
-"AR      ?= ar ru ;\n",
+"AR      ?= ar qf ;\n",
 "AS      ?= as ;\n",
 "ASFLAGS     += $(ASFLAGS) ;\n",
 "AWK     ?= awk ;\n",
--- ./Jambase~	2016-12-14 22:17:04.000000000 +0000
+++ ./Jambase	2016-12-14 22:19:30.498595843 +0000
@@ -837,7 +837,7 @@
 # General defaults; a lot like UNIX
 #
 
-    AR      ?= ar ru ;
+    AR      ?= ar qf ;
     AS      ?= as ;
     ASFLAGS     ?= ;
     AWK     ?= awk ;
