--- test/test.c.orig	2013-10-16 11:36:09.361055699 +0100
+++ test/test.c	2013-10-16 12:02:57.455386784 +0100
@@ -1475,6 +1475,72 @@
                     }
                     modulus = 2;
                 }
+		if (!strcasecmp(preset_name, "N900"))
+		{
+                    if( !mux )
+                    {
+                        mux = HB_MUX_MP4;
+                    }
+                    job->largeFileSize = 1;
+                    vcodec = HB_VCODEC_X264;
+                    job->vbitrate = 960;
+                    job->vquality = 20.0;
+                    if( !atracks )
+                    {
+                        atracks = strdup("1");
+                    }
+                    if( !acodecs )
+                    {
+                        acodecs = strdup("faac");
+                    }
+                    if( !abitrates )
+                    {
+                        abitrates = str_split("160", ',');
+                    }
+                    if( !mixdowns )
+                    {
+                        mixdowns = strdup("dpl2");
+                    }
+                    if( !arates )
+                    {
+                        arates = strdup("48000");
+                    }
+                    if( !dynamic_range_compression )
+                    {
+                        dynamic_range_compression = strdup("0.0");
+                    }
+                    if ( !maxWidth )
+                    {
+                        maxWidth = 800;
+                    }
+                    if ( !maxHeight )
+                    {
+                        maxHeight = 480;
+                    }
+                    if (x264_preset == NULL)
+                    {
+                        x264_preset = strdup("medium");
+                    }
+                    if (h264_profile == NULL)
+                    {
+                        h264_profile = strdup("baseline");
+                    }
+                    if (h264_level == NULL)
+                    {
+                        h264_level = strdup("3.0");
+                    }
+                    par_width = 1;
+                    par_height = 1;
+                    keep_display_aspect = 1;
+                    anamorphic_mode = 3;
+                    detelecine = 1;
+                    decomb = 1;
+                    deblock = 4;
+                    modulus = 16;
+                    mp4_optimize = 1;
+                    job->chapter_markers = 1;
+
+		}
                 if (!strcasecmp(preset_name, "Normal"))
                 {
                     if( !mux )
@@ -3602,6 +3668,7 @@
     printf("\n   + Android:  -e x264  -q 22.0 -r 30 --pfr  -a 1 -E faac -B 128 -6 dpl2 -R Auto -D 0.0 --audio-copy-mask aac,ac3,dtshd,dts,mp3 --audio-fallback ffac3 -f mp4 -X 720 -Y 576 --loose-anamorphic --modulus 2 --x264-preset medium --h264-profile main --h264-level 3.0\n");
     printf("\n   + Android Tablet:  -e x264  -q 22.0 -r 30 --pfr  -a 1 -E faac -B 128 -6 dpl2 -R Auto -D 0.0 --audio-copy-mask aac,ac3,dtshd,dts,mp3 --audio-fallback ffac3 -f mp4 -X 1280 -Y 720 --loose-anamorphic --modulus 2 --x264-preset medium --h264-profile main --h264-level 3.1\n");
     printf("\n   + Windows Phone 8:  -e x264  -q 22.0 -r 30 --pfr  -a 1 -E faac -B 128 -6 dpl2 -R Auto -D 0.0 --audio-copy-mask aac,ac3,dtshd,dts,mp3 --audio-fallback ffac3 -f mp4 -X 1280 -Y 720 --loose-anamorphic --modulus 2 --x264-preset medium --h264-profile main --h264-level 3.1\n");
+    printf("\n   + Nokia N900:  -e x264  -q 20.0 -a 1,1 -E faac -B 160,160 -6 dpl2,auto -R Auto,Auto -D 0.0 -f mp4 -4 -X 800 -Y 448 --custom-anamorphic -m -O --deblock 4 --modulus 16 -S 700 --x264-preset medium --h264-profile baseline --h264-level 3.0\n");
 	printf("\n>\n");
     printf("\n< Regular\n");
     printf("\n   + Normal:  -e x264  -q 20.0 -a 1 -E faac -B 160 -6 dpl2 -R Auto -D 0.0 --audio-copy-mask aac,ac3,dtshd,dts,mp3 --audio-fallback ffac3 -f mp4 --loose-anamorphic --modulus 2 -m --x264-preset veryfast --h264-profile main --h264-level 4.0\n");
