--- data/Makefile.am~	2015-06-17 08:00:42.000000000 +0100
+++ data/Makefile.am	2015-06-17 08:11:49.801847885 +0100
@@ -175,8 +175,6 @@
 
 systemdsystemunit =
 
-if HAVE_SYSTEMD
-
 gdm.service: $(srcdir)/gdm.service.in
 	$(AM_V_GEN)sed \
 		-e 's,[@]sbindir[@],$(sbindir),g' \
@@ -188,8 +186,6 @@
 systemdsystemunit += gdm.service
 CLEANFILES += gdm.service
 
-endif
-
 Xsession_files =
 if ENABLE_GDM_XSESSION
 
