--- ./Source/WebCore/platform/graphics/GLContext.h~	2017-01-07 09:12:11.000000000 +0000
+++ ./Source/WebCore/platform/graphics/GLContext.h	2017-01-07 10:47:52.894355625 +0000
@@ -25,7 +25,7 @@
 #include <wtf/Noncopyable.h>
 
 #if USE(EGL) && !PLATFORM(GTK)
-#include "eglplatform.h"
+#include <EGL/eglplatform.h>
 typedef EGLNativeWindowType GLNativeWindowType;
 #else
 typedef uint64_t GLNativeWindowType;
