From 2039097dde5b7f92246c52a1a87be8e4e779dd34 Mon Sep 17 00:00:00 2001
From: embar- <embar@super.lt>
Date: Sat, 29 Nov 2014 20:47:20 +0200
Subject: [PATCH] Update openct.udev.in

---
 etc/openct.udev.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/etc/openct.udev.in b/etc/openct.udev.in
index d11d0e1..a20d226 100644
--- a/etc/openct.udev.in
+++ b/etc/openct.udev.in
@@ -27,6 +27,9 @@ PROGRAM="/bin/sleep 0.1"
 # ccid
 ATTR{bInterfaceClass}=="0b", ATTR{bInterfaceSubClass}=="00", ATTR{bInterfaceProtocol}=="00", ATTRS{idVendor}=="?*"  RUN+="@udevdir@/openct_usb /dev/$parent"
 
+# Broadcom 5880
+ATTR{idVendor}=="0A5C", ATTR{idProduct}=="5800", RUN+="@udevdir@/openct_usb /dev/$name"
+ATTR{idVendor}=="0A5C", ATTR{idProduct}=="5801", RUN+="@udevdir@/openct_usb /dev/$name"
 # egate
 ATTR{idVendor}=="0973", ATTR{idProduct}=="0001", RUN+="@udevdir@/openct_usb /dev/$name"
 # eToken
