From dae1aebd3c95f6b07cdd3292929142d6f4d15e07 Mon Sep 17 00:00:00 2001
From: embar- <embar@super.lt>
Date: Sat, 29 Nov 2014 20:49:39 +0200
Subject: [PATCH] Update openct.udev.modalias.in

---
 etc/openct.udev.modalias.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/etc/openct.udev.modalias.in b/etc/openct.udev.modalias.in
index c0b1197..584f4cf 100644
--- a/etc/openct.udev.modalias.in
+++ b/etc/openct.udev.modalias.in
@@ -13,6 +13,9 @@
 SUBSYSTEM!="usb", GOTO="openct_usb_rules_end"
 ACTION!="add", GOTO="openct_usb_rules_end"
 
+# Broadcom 5880
+ENV{MODALIAS}=="usb:v0A5Cp5800*", RUN+="@udevdir@/openct_usb"
+ENV{MODALIAS}=="usb:v0A5Cp5801*", RUN+="@udevdir@/openct_usb"
 # egate
 ENV{MODALIAS}=="usb:v0973p0001*", RUN+="@udevdir@/openct_usb"
 # eToken
