From 7efa3d224f5fefdeb8467937ed6bf21ef7f69bcb Mon Sep 17 00:00:00 2001
From: embar- <embar@super.lt>
Date: Sat, 29 Nov 2014 20:52:09 +0200
Subject: [PATCH] Update openct.fdi

---
 etc/openct.fdi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/etc/openct.fdi b/etc/openct.fdi
index 0fec6df..7e129bb 100644
--- a/etc/openct.fdi
+++ b/etc/openct.fdi
@@ -3,6 +3,18 @@
   <device>
     <match key="info.subsystem" string="usb">
 
+      <match key="usb.vendor_id" int="0x0A5C">
+	<match key="usb.product_id" int="0x5800">
+	  <merge key="smart_card_reader.openct_capable" type="bool">true</merge>
+	</match>
+      </match>
+
+      <match key="usb.vendor_id" int="0x0A5C">
+	<match key="usb.product_id" int="0x5801">
+	  <merge key="smart_card_reader.openct_capable" type="bool">true</merge>
+	</match>
+      </match>
+
       <match key="usb.vendor_id" int="0x0973">
 	<match key="usb.product_id" int="0x0001">
 	  <merge key="smart_card_reader.openct_capable" type="bool">true</merge>
