From b8f3c7161a803c48844266d8de7cf4d957b56101 Mon Sep 17 00:00:00 2001
From: embar- <embar@super.lt>
Date: Sat, 29 Nov 2014 20:54:06 +0200
Subject: [PATCH] add Broadcom 5880 Smartcard/UPEK Reader

---
 etc/openct.conf.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/etc/openct.conf.in b/etc/openct.conf.in
index f4af393..1104be2 100644
--- a/etc/openct.conf.in
+++ b/etc/openct.conf.in
@@ -135,6 +135,8 @@ driver	ccid {
 		usb:0bf8/1006,	# fujitsu siemens 3.5" drive size reader
 		usb:0dc3/1004,	# Athena Smartcard Solutions, Inc. ASEKey
 		usb:0a89/0030,	# Aktiv Rutoken ECP
+		usb:0a5c/5800,  # Broadcom 5880 Smartcard/UPEK Reader
+		usb:0a5c/5801,  # Broadcom 5880 Smartcard/UPEK Reader
 	};
 };
 driver pertosmart1030 {
