--- src/backends/meta-input-settings.c~	2015-02-12 22:22:13.000000000 +0000
+++ src/backends/meta-input-settings.c	2015-02-13 08:14:35.513058055 +0000
@@ -600,6 +600,7 @@
   ClutterInputDeviceType type;
   GSettings *settings;
   gchar *path;
+  gint device_id;
 
   type = clutter_input_device_get_device_type (device);
 
@@ -619,8 +620,9 @@
   else
     return NULL;
 
-  vendor = clutter_input_device_get_vendor_id (device);
-  product = clutter_input_device_get_product_id (device);
+  device_id = clutter_input_device_get_device_id (device);
+  vendor = g_strdup_printf ("%u", device_id & 0xffff);
+  product = g_strdup_printf ("%u", device_id >> 16U);
   path = g_strdup_printf ("/org/gnome/desktop/peripherals/%s/%s:%s/",
                           group, vendor, product);
 
