--- src/compositor/meta-shaped-texture.c~	2014-12-08 10:42:55.000000000 +0000
+++ src/compositor/meta-shaped-texture.c	2014-12-08 10:56:04.040728174 +0000
@@ -394,7 +394,7 @@
 
   filter = COGL_PIPELINE_FILTER_LINEAR;
 
-  if (meta_actor_painting_untransformed (tex_width, tex_height, NULL, NULL))
+  if (meta_actor_painting_untransformed (dest_width, dest_height, NULL, NULL))
     filter = COGL_PIPELINE_FILTER_NEAREST;
 
   ctx = clutter_backend_get_cogl_context (clutter_get_default_backend ());
