--- hw/xfree86/common/xf86Bus.c~	2013-10-02 23:01:18.000000000 +0100
+++ hw/xfree86/common/xf86Bus.c	2013-10-02 23:19:09.100608444 +0100
@@ -81,7 +81,7 @@
     if (drv->platformProbe != NULL) {
         foundScreen = xf86platformProbeDev(drv);
     }
-    if (ServerIsNotSeat0())
+    if (ServerIsNotSeat0() && foundScreen)
         return foundScreen;
 #endif
 
@@ -201,7 +201,7 @@
 {
 #ifdef XSERVER_PLATFORM_BUS
     xf86platformProbe();
-    if (ServerIsNotSeat0())
+    if (ServerIsNotSeat0() && xf86_num_platform_devices > 0)
         return;
 #endif
 #ifdef XSERVER_LIBPCIACCESS
