--- ./third_party/angle/BUILD.gn.orig	2017-03-24 19:20:38.561670401 +0000
+++ ./third_party/angle/BUILD.gn	2017-03-24 19:23:29.583902664 +0000
@@ -125,6 +125,7 @@
 }
 
 config("angle_common_config") {
+  cflags = [ "-fPIC", ]
   include_dirs = [ "src/common/third_party/numerics" ]
 }
 
--- ./third_party/angle/src/third_party/libXNVCtrl/BUILD.gn~	2017-02-16 23:03:32.000000000 +0000
+++ ./third_party/angle/src/third_party/libXNVCtrl/BUILD.gn	2017-03-24 19:39:47.956661523 +0000
@@ -10,6 +10,7 @@
   # XextAddDisplay which will cause a failure.
   cflags = [
     "-Wno-incompatible-pointer-types-discards-qualifiers",
+    "-fPIC",
   ]
 }
 
--- ./third_party/angle/BUILD.gn~	2017-03-24 19:42:35.000000000 +0000
+++ ./third_party/angle/BUILD.gn	2017-03-24 19:52:38.950795120 +0000
@@ -570,6 +570,8 @@
     defines += [ "EGLAPI=__attribute__((visibility(\"default\")))" ]
   }
 
+  cflags = [ "-fPIC", ]
+
   if (is_win) {
     ldflags = [ "/DEF:" + rebase_path("src/libEGL/libEGL.def", root_build_dir) ]
   }
--- ./base/allocator/BUILD.gn~	2017-02-16 23:02:25.000000000 +0000
+++ ./base/allocator/BUILD.gn	2017-03-24 23:15:57.844567570 +0000
@@ -86,6 +86,7 @@
       "-Wno-unused-result",
     ]
   }
+  cflags += [ "-fPIC", ]
 }
 
 # This config is only used on Windows static release builds for the
--- ./base/third_party/dynamic_annotations/BUILD.gn~	2017-02-16 23:02:25.000000000 +0000
+++ ./base/third_party/dynamic_annotations/BUILD.gn	2017-03-25 09:08:34.001966936 +0000
@@ -23,5 +23,6 @@
       configs -= [ "//build/config/compiler:default_optimization" ]
       configs += [ "//build/config/compiler:optimize_max" ]
     }
+    cflags = [ "-fPIC", ]
   }
 }
--- ./base/BUILD.gn~	2017-02-16 23:02:25.000000000 +0000
+++ ./base/BUILD.gn	2017-03-25 09:23:12.829441598 +0000
@@ -51,8 +51,9 @@
 }
 
 config("base_flags") {
+  cflags = [ "-fPIC", ]
   if (is_clang) {
-    cflags = [
+    cflags += [
       # Don't die on dtoa code that uses a char as an array index.
       # This is required solely for base/third_party/dmg_fp/dtoa_wrapper.cc.
       "-Wno-char-subscripts",
--- ./third_party/widevine/cdm/BUILD.gn~	2017-02-16 23:02:55.000000000 +0000
+++ ./third_party/widevine/cdm/BUILD.gn	2017-03-25 10:02:52.072634602 +0000
@@ -109,6 +109,7 @@
       "//build/config/sanitizers:deps",
     ]
 
+    cflags = [ "-fPIC", ]
     if (is_mac) {
       ldflags = [
         # Not to strip important symbols by -Wl,-dead_strip.
@@ -118,7 +119,7 @@
       ]
       #TODO(jrummell) Mac: 'DYLIB_INSTALL_NAME_BASE': '@loader_path',
     } else if (is_posix && !is_mac) {
-      cflags = [ "-fvisibility=hidden" ]
+      cflags += [ "-fvisibility=hidden" ]
     }
   }
 } else {
@@ -165,6 +165,7 @@
       ":widevinecdm",
     ]
 
+    cflags = [ "-fPIC", ]
     if (is_linux) {
       ldflags =
           [ rebase_path("$root_out_dir/$widevine_cdm_path/libwidevinecdm.so",
--- ./ppapi/cpp/BUILD.gn~	2017-02-16 23:02:36.000000000 +0000
+++ ./ppapi/cpp/BUILD.gn	2017-03-25 10:23:36.358203815 +0000
@@ -37,13 +37,13 @@
       configs -= [ "//build/config/gcc:symbol_visibility_hidden" ]
       configs += [ "//build/config/gcc:symbol_visibility_default" ]
 
-      cflags = [ "-fPIC" ]
     } else {
       # This library is distributed as a part of the SDK and as such has to
       # be a static library rather than a source set.
       complete_static_lib = true
     }
   }
+  cflags = [ "-fPIC" ]
 }
 
 # Link to this target to get only the PPAPI C++ wrapper objects but not the
@@ -301,7 +301,7 @@
       configs -= [ "//build/config/gcc:symbol_visibility_hidden" ]
       configs += [ "//build/config/gcc:symbol_visibility_default" ]
 
-      cflags = [ "-fPIC" ]
     }
   }
+  cflags = [ "-fPIC" ]
 }
--- ./v8/BUILD.gn~	2017-02-16 23:03:42.000000000 +0000
+++ ./v8/BUILD.gn	2017-03-25 13:12:00.707941677 +0000
@@ -413,6 +413,7 @@
       cflags += [ "-Wshorten-64-to-32" ]
     }
   }
+  cflags += [ "-fPIC", ]
 }
 
 ###############################################################################
--- ./chrome/common/BUILD.gn~	2017-03-25 13:46:48.000000000 +0000
+++ ./chrome/common/BUILD.gn	2017-03-25 13:47:34.511806554 +0000
@@ -533,6 +533,7 @@
     ]
     public_deps += [ "//chrome/common:version_header" ]
   }
+  cflags = [ "-fPIC", ]
 }
 
 process_version("version_header") {
