diff -urN chromium-37.0.2062.0.orig/ui/gl/BUILD.gn chromium-37.0.2062.0/ui/gl/BUILD.gn
--- chromium-37.0.2062.0.orig/ui/gl/BUILD.gn	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/BUILD.gn	2014-07-19 13:43:37.861645482 +0100
@@ -103,8 +103,6 @@
 
   include_dirs = [
     "//third_party/switfshader/include",
-    "//third_party/khronos",
-    "//third_party/mesa/src/include",
     gl_binding_output_dir,
   ]
 
@@ -115,7 +113,6 @@
     "//base",
     "//base/third_party/dynamic_annotations",
     "//gpu/command_buffer/common",
-    "//third_party/mesa:mesa_headers",
     "//skia",
     "//ui/events/platform",
     "//ui/gfx",
@@ -123,7 +120,7 @@
   ]
 
   # TODO(GYP) hook up once this is converted.
-  forward_dependent_configs_from = [ "//third_party/mesa:mesa_headers" ]
+  #forward_dependent_configs_from = [ "//third_party/mesa:mesa_headers" ]
 
   if (is_win || is_android || is_linux) {
     sources += [
@@ -260,14 +257,9 @@
   source_prereqs = [
     "EGL/eglextchromium.h",
     "GL/glextchromium.h",
-    "//third_party/mesa/src/include/GL/glext.h",
-    "//third_party/khronos/GLES2/gl2ext.h",
     "//gpu/GLES2/gl2chromium.h",
     "//gpu/GLES2/gl2extchromium.h",
     "//third_party/khronos/EGL/eglext.h",
-    "//third_party/mesa/src/include/GL/wglext.h",
-    "//third_party/mesa/src/include/GL/glx.h",
-    "//third_party/mesa/src/include/GL/glxext.h",
   ]
 
   outputs = [
@@ -293,8 +285,6 @@
 
   args = [
     "--header-paths=" +
-        rebase_path("//third_party/khronos", root_build_dir) + ":" +
-        rebase_path("//third_party/mesa/src/include", root_build_dir) + ":" +
         rebase_path("//ui/gl", root_build_dir) + ":" +
         rebase_path("//gpu", root_build_dir),
     rebase_path(gl_binding_output_dir, root_build_dir),
diff -urN chromium-37.0.2062.0.orig/ui/gl/DEPS chromium-37.0.2062.0/ui/gl/DEPS
--- chromium-37.0.2062.0.orig/ui/gl/DEPS	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/DEPS	2014-07-19 13:43:54.494317438 +0100
@@ -1,6 +1,4 @@
 include_rules = [
-  "+third_party/khronos",
-  "+third_party/mesa/src/include",
   "+third_party/skia",
   "+ui/events",
   "+ui/gfx",
diff -urN chromium-37.0.2062.0.orig/ui/gl/egl_util.cc chromium-37.0.2062.0/ui/gl/egl_util.cc
--- chromium-37.0.2062.0.orig/ui/gl/egl_util.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/egl_util.cc	2014-07-19 13:39:28.205570233 +0100
@@ -7,7 +7,7 @@
 #if defined(OS_ANDROID)
 #include <EGL/egl.h>
 #else
-#include "third_party/khronos/EGL/egl.h"
+#include <EGL/egl.h>
 #endif
 
 // This needs to be after the EGL includes
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl.gyp chromium-37.0.2062.0/ui/gl/gl.gyp
--- chromium-37.0.2062.0.orig/ui/gl/gl.gyp	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl.gyp	2014-07-19 13:46:13.545301733 +0100
@@ -17,7 +17,6 @@
         '<(DEPTH)/base/third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
         '<(DEPTH)/gpu/command_buffer/command_buffer.gyp:gles2_utils',
         '<(DEPTH)/skia/skia.gyp:skia',
-        '<(DEPTH)/third_party/mesa/mesa.gyp:mesa_headers',
         '<(DEPTH)/ui/gfx/gfx.gyp:gfx',
         '<(DEPTH)/ui/gfx/gfx.gyp:gfx_geometry',
       ],
@@ -29,8 +28,6 @@
       ],
       'include_dirs': [
         '<(DEPTH)/third_party/swiftshader/include',
-        '<(DEPTH)/third_party/khronos',
-        '<(DEPTH)/third_party/mesa/src/include',
         '<(gl_binding_output_dir)',
       ],
       'direct_dependent_settings': {
@@ -38,9 +35,6 @@
           '<(gl_binding_output_dir)',
         ],
       },
-      'export_dependent_settings': [
-        '<(DEPTH)/third_party/mesa/mesa.gyp:mesa_headers',
-      ],
       'sources': [
         'android/gl_jni_registrar.cc',
         'android/gl_jni_registrar.h',
@@ -135,8 +129,7 @@
           'action_name': 'generate_gl_bindings',
           'variables': {
             'generator_path': 'generate_bindings.py',
-            # Prefer khronos EGL/GLES headers by listing that path first.
-            'header_paths': '../../third_party/khronos:../../third_party/mesa/src/include:.:../../gpu',
+            'header_paths': '/usr/include:.:../../gpu',
           },
           'inputs': [
             '<(generator_path)',
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_context_egl.cc chromium-37.0.2062.0/ui/gl/gl_context_egl.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_context_egl.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_context_egl.cc	2014-07-19 13:39:28.216571337 +0100
@@ -8,8 +8,8 @@
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
 #include "build/build_config.h"
-#include "third_party/khronos/EGL/egl.h"
-#include "third_party/khronos/EGL/eglext.h"
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
 #include "ui/gl/egl_util.h"
 #include "ui/gl/gl_bindings.h"
 #include "ui/gl/gl_surface_egl.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_context_glx.cc chromium-37.0.2062.0/ui/gl/gl_context_glx.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_context_glx.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_context_glx.cc	2014-07-19 13:39:28.235573244 +0100
@@ -11,7 +11,7 @@
 #include "base/debug/trace_event.h"
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/gl/GL/glextchromium.h"
 #include "ui/gl/gl_bindings.h"
 #include "ui/gl/gl_implementation.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_context_mac.mm chromium-37.0.2062.0/ui/gl/gl_context_mac.mm
--- chromium-37.0.2062.0.orig/ui/gl/gl_context_mac.mm	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_context_mac.mm	2014-07-19 13:39:28.192568928 +0100
@@ -6,7 +6,7 @@
 #include "base/debug/trace_event.h"
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/gl/gl_context_cgl.h"
 #include "ui/gl/gl_context_osmesa.h"
 #include "ui/gl/gl_context_stub.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_android.cc chromium-37.0.2062.0/ui/gl/gl_surface_android.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_android.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_android.cc	2014-07-19 13:39:28.224572140 +0100
@@ -6,7 +6,7 @@
 
 #include "base/logging.h"
 #include "base/memory/ref_counted.h"
-#include "third_party/khronos/EGL/egl.h"
+#include <EGL/egl.h>
 #include "ui/gfx/native_widget_types.h"
 #include "ui/gl/gl_implementation.h"
 #include "ui/gl/gl_surface_egl.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_egl.cc chromium-37.0.2062.0/ui/gl/gl_surface_egl.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_egl.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_egl.cc	2014-07-19 13:39:28.231572842 +0100
@@ -4,7 +4,7 @@
 
 // This include must be here so that the includes provided transitively
 // by gl_surface_egl.h don't make it impossible to compile this code.
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 
 #include "ui/gl/gl_surface_egl.h"
 
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_glx.cc chromium-37.0.2062.0/ui/gl/gl_surface_glx.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_glx.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_glx.cc	2014-07-19 13:39:28.218571538 +0100
@@ -20,7 +20,7 @@
 #include "base/threading/non_thread_safe.h"
 #include "base/threading/thread.h"
 #include "base/time/time.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/events/platform/platform_event_source.h"
 #include "ui/gfx/x/x11_connection.h"
 #include "ui/gfx/x/x11_types.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_mac.cc chromium-37.0.2062.0/ui/gl/gl_surface_mac.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_mac.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_mac.cc	2014-07-19 13:39:28.215571237 +0100
@@ -11,7 +11,7 @@
 #include "base/logging.h"
 #include "base/mac/mac_util.h"
 #include "base/memory/scoped_ptr.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/gl/gl_bindings.h"
 #include "ui/gl/gl_context.h"
 #include "ui/gl/gl_implementation.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_osmesa.cc chromium-37.0.2062.0/ui/gl/gl_surface_osmesa.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_osmesa.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_osmesa.cc	2014-07-19 13:39:28.230572742 +0100
@@ -3,7 +3,7 @@
 // found in the LICENSE file.
 
 #include "base/logging.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/gl/gl_bindings.h"
 #include "ui/gl/gl_context.h"
 #include "ui/gl/gl_surface_osmesa.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_win.cc chromium-37.0.2062.0/ui/gl/gl_surface_win.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_win.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_win.cc	2014-07-19 13:39:28.236573344 +0100
@@ -11,7 +11,7 @@
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
 #include "base/win/windows_version.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/gfx/frame_time.h"
 #include "ui/gfx/native_widget_types.h"
 #include "ui/gl/gl_bindings.h"
diff -urN chromium-37.0.2062.0.orig/ui/gl/gl_surface_x11.cc chromium-37.0.2062.0/ui/gl/gl_surface_x11.cc
--- chromium-37.0.2062.0.orig/ui/gl/gl_surface_x11.cc	2014-06-21 08:02:40.000000000 +0100
+++ chromium-37.0.2062.0/ui/gl/gl_surface_x11.cc	2014-07-19 13:39:28.209570634 +0100
@@ -8,7 +8,7 @@
 #include "base/logging.h"
 #include "base/memory/scoped_ptr.h"
 #include "base/message_loop/message_loop.h"
-#include "third_party/mesa/src/include/GL/osmesa.h"
+#include <GL/osmesa.h>
 #include "ui/gfx/native_widget_types.h"
 #include "ui/gfx/x/x11_types.h"
 #include "ui/gl/gl_bindings.h"
