--- chromium-50.0.2661.57/third_party/webrtc/build/common.gypi~	2016-03-30 20:04:01.000000000 +0100
+++ chromium-50.0.2661.57/third_party/webrtc/build/common.gypi	2016-04-05 22:29:45.419157347 +0100
@@ -153,7 +153,7 @@
       # initialize. See also: |rtc_initialize_ffmpeg|.
       # CHECK THE OPENH264, FFMPEG AND H.264 LICENSES/PATENTS BEFORE BUILDING.
       # http://www.openh264.org, https://www.ffmpeg.org/
-      ['proprietary_codecs==1 and OS!="android" and OS!="ios"', {
+      ['proprietary_codecs==1 and OS!="android" and OS!="ios" and OS!="linux"', {
         'rtc_use_h264%': 1,
       }, {
         'rtc_use_h264%': 0,
--- chromium-50.0.2661.57/third_party/webrtc/build/webrtc.gni~	2016-03-30 20:04:01.000000000 +0100
+++ chromium-50.0.2661.57/third_party/webrtc/build/webrtc.gni	2016-04-05 22:33:34.752236803 +0100
@@ -100,7 +100,7 @@
   # also: |rtc_initialize_ffmpeg|.
   # CHECK THE OPENH264, FFMPEG AND H.264 LICENSES/PATENTS BEFORE BUILDING.
   # http://www.openh264.org, https://www.ffmpeg.org/
-  rtc_use_h264 = proprietary_codecs && !is_android && !is_ios
+  rtc_use_h264 = proprietary_codecs && !is_android && !is_ios && !is_linux
 
   # FFmpeg must be initialized for |H264DecoderImpl| to work. This can be done
   # by WebRTC during |H264DecoderImpl::InitDecode| or externally. FFmpeg must
