--- ./third_party/WebKit/Source/BUILD.gn~	2016-03-29 20:02:03.000000000 +0100
+++ ./third_party/WebKit/Source/BUILD.gn	2016-04-06 00:07:13.292683604 +0100
@@ -92,9 +92,9 @@
   }
 
   ## TODO(GYP) : gn does not yet support use_system_icu.
-  #if (use_system_icu) {
-  #  defines += [ "USING_SYSTEM_ICU" ]
-  #}
+  if (use_system_icu) {
+    defines += [ "USING_SYSTEM_ICU" ]
+  }
 }
 
 # The follow configs apply to all targets except for unit tests, which rely on
--- ./build/config/BUILD.gn~	2016-03-29 20:01:56.000000000 +0100
+++ ./build/config/BUILD.gn	2016-04-06 15:39:05.094121165 +0100
@@ -145,6 +145,9 @@
   if (use_allocator != "tcmalloc") {
     defines += [ "NO_TCMALLOC" ]
   }
+  if (use_system_icu) {
+    defines += [ "USING_SYSTEM_ICU" ]
+  }
   if (is_asan || is_lsan || is_tsan || is_msan) {
     defines += [
       "MEMORY_TOOL_REPLACES_ALLOCATOR",
--- ./build/common.gypi~	2016-03-29 20:01:56.000000000 +0100
+++ ./build/common.gypi	2016-04-06 15:40:38.090684882 +0100
@@ -2768,6 +2768,9 @@
       ['use_x11==1', {
         'defines': ['USE_X11=1'],
       }],
+      ['use_system_icu==1', {
+        'defines': ['USING_SYSTEM_ICU=1'],
+      }],
       ['use_clipboard_aurax11==1', {
         'defines': ['USE_CLIPBOARD_AURAX11=1'],
       }],
