--- src/AudioSDLMixer.cpp~	2014-07-16 18:16:58.000000000 +0100
+++ src/AudioSDLMixer.cpp	2014-07-16 18:27:15.920748162 +0100
@@ -93,7 +93,7 @@
 }
 
 /**
- * pause music channel. If CDROM enabled, call Audio::pauseGameMusic();
+ * pause music channel.
  */
 //----------------------------------------------------------
 void AudioSDLMixer::pauseGameMusic(bool status)
@@ -101,22 +101,15 @@
 	Config	*config = Config::instance();
 	if (config->audioEnabled()) 
 	{
-		if(cdrom)
-		{
-			Audio::pauseGameMusic(status);
-		}
-		else
-		{
 	    	if (status)
 				Mix_PauseMusic();
 			else
 				Mix_ResumeMusic();
-		}
 	}
 }
 
 /**
- * stop music channel. If CDROM enabled, call Audio::stopMusic();
+ * stop music channel.
  */
 //----------------------------------------------------------
 void	AudioSDLMixer::stopMusic()
@@ -140,10 +133,7 @@
 		{
 			default:
 			case MusicGame:
-				if(cdrom)
-					Mix_HaltChannel (0);
-				else
-					Mix_PlayChannel (0, sounds[mode], -1);
+                                Mix_PlayChannel (0, sounds[mode], -1);
 				break;
 			case MusicMenu:
 				Mix_PlayChannel (0, sounds[mode], -1);
