--- embed/ephy-embed-shell.c~	2014-10-22 16:04:12.000000000 +0100
+++ embed/ephy-embed-shell.c	2014-10-22 16:43:39.096299235 +0100
@@ -525,12 +525,9 @@
 static void
 ephy_embed_shell_setup_web_extensions_connection (EphyEmbedShell *shell)
 {
-  GError *error = NULL;
-
   shell->priv->bus = g_application_get_dbus_connection (G_APPLICATION (shell));
   if (!shell->priv->bus) {
-    g_warning ("Application not connected to session bus: %s\n", error->message);
-    g_error_free (error);
+    g_warning ("Application not connected to session bus!\n");
     return;
   }
 
