--- src/Makefile~	2015-05-28 20:22:50.000000000 +0100
+++ src/Makefile	2015-05-28 20:31:04.201675973 +0100
@@ -79,6 +79,7 @@
 spidermonkey/js/src: spidermonkey/js-?.?.?.tar.gz
 	tar xzvf spidermonkey/js-?.?.?.tar.gz -C spidermonkey
 	sed -i -e 's:\(shell uname -s | sed /\\ /s//_/\):\1 | sed s,GNU.*,Linux,:g' spidermonkey/js/src/config.mk
+	sed -i -e 's/_BSD_SOURCE/_DEFAULT_SOURCE/g' spidermonkey/js/src/config/Linux*.mk
 
 jsapi_buildstamp: spidermonkey/js/src
 	cd spidermonkey && SMCFLAGS="$(SHFLAGS) $(SMCFLAGS)" $(MAKE) jsapi
