--- source4/kdc/kdc.c~	2014-10-01 10:16:21.000000000 +0100
+++ source4/kdc/kdc.c	2015-03-12 22:46:27.240052144 +0000
@@ -967,9 +967,9 @@
 	 * The old behavior in the _kdc_get_preferred_key()
 	 * function is use_strongest_server_key=TRUE.
 	 */
-	kdc->config->as_use_strongest_session_key = false;
+	kdc->config->tgt_use_strongest_session_key = false;
 	kdc->config->preauth_use_strongest_session_key = false;
-	kdc->config->tgs_use_strongest_session_key = false;
+	kdc->config->svc_use_strongest_session_key = false;
 	kdc->config->use_strongest_server_key = true;
 
 	/* Register hdb-samba4 hooks for use as a keytab */
