--- ./src/command/StickerCommands.cxx~	2016-10-02 17:57:06.000000000 +0100
+++ ./src/command/StickerCommands.cxx	2016-10-02 18:02:50.922026647 +0100
@@ -33,7 +33,7 @@
 #include "util/ConstBuffer.hxx"
 #include "util/StringAPI.hxx"
 
-struct sticker_song_find_data {
+struct _sticker_song_find_data {
 	Response &r;
 	Partition &partition;
 	const char *name;
@@ -43,8 +43,8 @@
 sticker_song_find_print_cb(const LightSong &song, const char *value,
 			   void *user_data)
 {
-	struct sticker_song_find_data *data =
-		(struct sticker_song_find_data *)user_data;
+	struct _sticker_song_find_data *data =
+		(struct _sticker_song_find_data *)user_data;
 
 	song_print_uri(data->r, data->partition, song);
 	sticker_print_value(data->r, data->name, value);
@@ -157,7 +157,7 @@
 			}
 		}
 
-		struct sticker_song_find_data data = {
+		struct _sticker_song_find_data data = {
 			r,
 			partition,
 			args[3],
