--- a/lib-python/2.7/distutils/command/install.py
+++ b/lib-python/2.7/distutils/command/install.py
@@ -87,6 +87,13 @@
         'purelib': '$base/site-packages',
         'platlib': '$base/site-packages',
         'headers': '$base/include',
+        'scripts': '/usr/bin',
+        'data'   : '$base',
+        },
+    'pypy_prefix': {
+        'purelib': '$base/site-packages',
+        'platlib': '$base/site-packages',
+        'headers': '$base/include',
         'scripts': '$base/bin',
         'data'   : '$base',
         },
@@ -474,9 +481,11 @@
 
     def select_scheme (self, name):
         # it's the caller's problem if they supply a bad name!
-        if (hasattr(sys, 'pypy_version_info') and
-            not name.endswith(('_user', '_home'))):
-            name = 'pypy'
+        if hasattr(sys, 'pypy_version_info'):
+            if name.endswith('_prefix'):
+                name = 'pypy_prefix'
+            elif not name.endswith(('_user', '_home')):
+                name = 'pypy'
         scheme = INSTALL_SCHEMES[name]
         for key in SCHEME_KEYS:
             attrname = 'install_' + key
