--- rpython/translator/c/gcc/trackgcroot.py~	2015-10-02 09:36:03.000000000 +0100
+++ rpython/translator/c/gcc/trackgcroot.py	2015-10-02 11:52:47.408494245 +0100
@@ -1072,6 +1072,7 @@
     visit_leaq = FunctionGcRootTracker._visit_lea
 
     visit_xorq = FunctionGcRootTracker.binary_insn
+    visit_xchgb = FunctionGcRootTracker._visit_xchg
     visit_xchgl = FunctionGcRootTracker._visit_xchg
     visit_xchgq = FunctionGcRootTracker._visit_xchg
     visit_testq = FunctionGcRootTracker._visit_test
