--- Source/cmake/FindOpenGL.cmake~	2015-06-22 09:58:36.000000000 +0100
+++ Source/cmake/FindOpenGL.cmake	2015-07-06 23:09:37.792292374 +0100
@@ -57,8 +57,10 @@
     # have OPENGL_INCLUDE_DIRS as part of the directories check_include_files() looks for in
     # case OpenGL is installed into a non-standard location.
     include(CMakePushCheckState)
-    CMAKE_PUSH_CHECK_STATE()
-    set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${OPENGL_INCLUDE_DIRS})
-    check_include_files("GL/glx.h" GLX_FOUND)
-    CMAKE_POP_CHECK_STATE()
+    if (ENABLE_GLX)
+        CMAKE_PUSH_CHECK_STATE()
+        set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} ${OPENGL_INCLUDE_DIR})
+        check_include_files("GL/glx.h" GLX_FOUND)
+        CMAKE_POP_CHECK_STATE()
+    endif ()
 endif ()
